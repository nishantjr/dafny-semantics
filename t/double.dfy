method Verify(n : int) returns (r : int)
  requires true
  ensures n * 2 == r
{
  r := n * 2 ;
}

method Main()
  returns (r: int)
  requires true
  ensures  true
{
  assert(Verify(0) == 0);
  assert(Verify(10) == 20);
  assert(Verify(-10) == -20);
}
