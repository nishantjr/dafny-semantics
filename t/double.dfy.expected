{
  Result
#Equals
  <generatedTop>
    <k>
      .
    </k>
    <methods>
      Main |-> method Main ( .ArgDecls ) returns ( r : int , .ArgDecls ) requires true ensures true { assert Verify ( 0 , .ResultExps ) == 0 ;  assert Verify ( 10 , .ResultExps ) == 20 ;  assert Verify ( -10 , .ResultExps ) == -20 ;  .Statements }
      Verify |-> method Verify ( n : int , .ArgDecls ) returns ( r : int , .ArgDecls ) requires true ensures n * 2 == r { r := n * 2 ;  .Statements }
    </methods>
    <stack>
      .List
    </stack>
    <store>
      0 |-> ( _0:Int , var )
      1 |-> ( _0 *Int 2 , var )
    </store>
    <env>
      n |-> 0
      r |-> 1
    </env>
    <nextLoc>
      2
    </nextLoc>
  </generatedTop>
}
